src/shared/unit-file.c

```diff
@@ -53,19 +51,35 @@ int unit_validate_alias_symlink_and_warn(const char *filename, const char *targe
                                         "%s: templates not allowed for %s units, rejecting.",
                                         filename, unit_type_to_string(src_type));
 
-        dst_name_type = unit_name_classify(dst);
+        dst_name_type = unit_name_to_instance(dst, &dst_instance);
         if (dst_name_type < 0)
                 return log_notice_errno(dst_name_type == -EINVAL ? SYNTHETIC_ERRNO(EXDEV) : dst_name_type,
                                         "%s points to \"%s\" which is not a valid unit name: %m",
                                         filename, dst);
-        if (dst_name_type == UNIT_NAME_INSTANCE)
-                return log_notice_errno(SYNTHETIC_ERRNO(EXDEV),
-                                        "%s: unit symlink target \"%s\" has instance name, rejecting.",
-                                        filename, dst);
-        if (dst_name_type != src_name_type)
+
+        if (dst_name_type == UNIT_NAME_INSTANCE) {
+                if (src_name_type != UNIT_NAME_INSTANCE)
+                        return log_notice_errno(SYNTHETIC_ERRNO(EXDEV),
+                                                "%s: unit symlink target \"%s\" has instance name, rejecting.",
+                                                filename, dst);
```

Comment:
this error doesn't look right... also, we could probably drop this, if we swap the order of the next if block with this one